src/jslisten.c

```diff
@@ -537,28 +592,30 @@ int bindJoy(void) {
 
   // Non-blocking reading
   struct js_event js;
-  int needTrigger;
+  int triggerSection;
   int lastTriggeredSet = -1;
   fcntl(joyFD, F_SETFL, O_NONBLOCK);
 
   while (1) {
-    while (read(joyFD, &js, sizeof(struct js_event)) == sizeof(struct js_event))  {
+    while (read(joyFD, &js, sizeof(struct js_event)) == sizeof(struct js_event) || delayedSection >= 0) {
```

Comment:
if `delayedSection` is true button checker loop continues till either
- delay button is released
- macro/section event triggers